cmd/zqd/listen/command.go

```diff
@@ -34,5 +37,30 @@ func New(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
 }
 
 func (c *Command) Run(args []string) error {
-	return zqd.Run(c.listenAddr)
+	logger := newLogger()
+	handler := zqd.NewHandler()
+	ln, err := net.Listen("tcp", c.listenAddr)
+	if err != nil {
+		return err
+	}
+	logger.Info("Listening", zap.String("addr", ln.Addr().String()))
+	return http.Serve(ln, handler)
+}
+
+func newLogger() *zap.Logger {
+	encoder := zap.NewProductionEncoderConfig()
+	encoder.CallerKey = ""
+	c := zap.Config{
+		Level:            zap.NewAtomicLevelAt(zap.DebugLevel),
+		Development:      false,
+		Encoding:         "json",
+		EncoderConfig:    encoder,
+		OutputPaths:      []string{"stdout"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
```

Comment:
Maybe shorten this since it's the production config with a few tweaks?
```go
c := zap.NewProductionConfig()
c.Level = zap.NewAtomicLevelAt(zap.DebugLevel)
c.Sampling = nil
c.EncoderConfig.CallerKey = ""
```

Reply:
They're both ugly but ok.